Build/Test Tools: Various improvements to `Tests_Theme_WpGetGlobalStylesheet`.

costdev opened the pull request:
This PR:

- [x] Converts the class to extend `WP_Theme_UnitTestCase`.
- [x] Adds a `@covers` annotation.
- [x] Renames the test class to be compliant with standards.
- [x] Implements a data provider for the tests to reduce code repetition.

Trac ticket:
https://core.trac.wordpress.org/ticket/57841
https://core.trac.wordpress.org/ticket/56793

Commit:
Add `@covers` annotation.

Commit:
Use a data provider for the tests.

Commit:
Convert to use `WP_Theme_UnitTestCase`.

Commit:
Make test class name compliant with standards.

hellofromtonya commented on tests/phpunit/tests/theme/wpEnqueueStoredStyles.php:
```diff
@@ -0,0 +1,74 @@
+<?php
```
This change for adding the test class and relocating its tests to it is ready for commit 👍 

I'll commit this separately from the improvement changes in `wpGetGLobalStylesheet.php`.

hellofromtonya replied:
Created https://github.com/WordPress/wordpress-develop/pull/4252 as a confidence check for commit.

hellofromtonya replied:
Committed via https://core.trac.wordpress.org/changeset/55567.

hellofromtonya replied:
@costdev I'll rebase your PR to bring in that commit. In doing so, this PR will change scope to only the  `wpGetGLobalStylesheet.php` improvements.

hellofromtonya replied:
Done ✅

hellofromtonya commented on tests/phpunit/tests/theme/wpEnqueueStoredStyles.php:
```diff
@@ -0,0 +1,74 @@
+/**
+ * Tests wp_enqueue_stored_styles().
+ *
+ * @group dependencies
+ * @group scripts
```
I don't think these groups are needed. While the function lives in the `script-loader.php` and uses Styles from the Dependencies API, this function is for the themes group. Removing these 2 groups from the commit.

hellofromtonya committed:
Reduces code in test methods.

1. Moves setup takes to helper methods.

2. Moves teardown tasks to the `tear_down()` method.

3. Splits the should and shouldn't tests into 2 separate tests.

4. Removes the function name from the test and data provider names.

5. Moves the expected CSS to the datasets in the data provider.
Why?
  * To keep the test data together in one place.
  * To allow for different themes with different font sizes to be added in the future.

hellofromtonya reviewed (approved):
I like the changes you did @costdev. I did some refactoring in [this commit](https://github.com/WordPress/wordpress-develop/pull/3745/commits/2f1a1c4648e01fffcf02455ed3dbabe5393d54d7) to help with readability and maintainability (what and why are in the commit message).

All of the existing tests are in this refactoring, most in the data providers as datasets rather than individual test methods.

Ready for commit 👍

hellofromtonya committed:
Updates docblock

hellofromtonya commented on tests/phpunit/tests/theme/wpGetGlobalStylesheet.php:
```diff
@@ -1,216 +1,264 @@
-		remove_theme_support( 'editor-font-sizes' );
+		$this->maybe_switch_theme( 'block-theme' );
+		$stylesheet_for_block_theme = wp_get_global_stylesheet();
+		$this->assertStringContainsString( '--wp--preset--font-size--custom: 100px;', $stylesheet_for_block_theme, 'custom font size (100px) is present for block theme' );
```
```suggestion
		$this->assertStringContainsString( $expected, $stylesheet_for_block_theme, 'Custom font size (100px) should be present for block theme' );
```

hellofromtonya commented on tests/phpunit/tests/theme/wpGetGlobalStylesheet.php:
```diff
@@ -1,216 +1,264 @@
-		$this->assertStringNotContainsString( '--wp--preset--font-size--large: 26.25px', $styles, 'large font size is not present' );
-		$this->assertStringNotContainsString( '--wp--preset--font-size--x-large: 42px', $styles, 'x-large font size is not present' );
+		$stylesheet_for_default_theme = wp_get_global_stylesheet();
+		$this->assertStringNotContainsString( $expected, $stylesheet_for_default_theme, 'custom font size (100px) not present for default theme' );
```
```suggestion
		$this->assertStringNotContainsString( $expected, $stylesheet_for_default_theme, 'Custom font size (100px) should not present for default theme' );
```

hellofromtonya committed:
Improves assertion messages. Uses $expected.

hellofromtonya commented:
Committed via https://core.trac.wordpress.org/changeset/55572.
